### DISCUSSION TITLE
Using the docker distribution and bootstrapping with no source checked out initially.

### DISCUSSION THREAD

**Discussion by Fhat** [oracle/opengrok#4174] [Q&A]
Hi, I'm trying to use the prebuilt docker image to run OpenGrok. We have some other configuration elsewhere that determines what gets checked out into the container's source directory ready for indexing. However, we'd like to not manually configure any projects, and just have OpenGrok configure projects based on what it sees in the source directory. It seems like this doesn't work if you're starting from a totally empty grok due to this early return here: https://github.com/oracle/opengrok/blob/master/docker/start.py#L223 #2734 

If I fork the file and swap the return for `webapp_projects = []`, the project refresh will continue happily and bootstrap the projects from the checked out code. 

I'm afraid I'm not super familiar with the codebase, and I'm sure the early return was added for a reason. I just wanted to check if I'm missing some configuration that would enable me to do this without having the change to the start.py script, or if not, is this a change we should look at raising a PR for? By the blame history @vladak seems like they'd be the most knowledgeable on this, but any input is more than appreciated.

Thanks!

**Comment by vladak** [comment]
That's a bug, I think. `list_projects()` is supposed to return list of projects by REST API call to the webapp. If it fails, it should return `None`. The caller should check for `None` specifically because empty list evaluates to false.
